backbone/src/main/java/org/researchstack/backbone/PermissionRequestManager.java

```diff
@@ -0,0 +1,293 @@
+package org.researchstack.backbone;
+import android.Manifest;
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.DrawableRes;
+import android.support.annotation.StringRes;
+import android.support.v4.content.ContextCompat;
+
+import org.researchstack.backbone.ui.step.layout.PermissionStepLayout;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is responsible for checking permissions during runtime. It is flexible enough to
+ * handle system and custom permissions (like gainging auth for a 3rd party service).
+ * <p>
+ * Any PermissionRequest should be added to the PermissionRequestManager during {@link
+ * Application#onCreate()}
+ * <p>
+ * The PermissionRequest objects are then presented in {@link PermissionStepLayout} for denile /
+ * approval
+ */
+public abstract class PermissionRequestManager
```

Comment:
Hmmm, I don't like how the history left this file in particular.  I've always had trouble when moving a file around and changing its contents in the same PR.  I'll try and fix this.